### PR TITLE
FHAC-644: Fix DS,DS Deferred Request and DS Deferred Response to do a endpoint URL lookup from Outbound classes

### PR DESCRIPTION
**Pull request by mhpnguyen**
Apply the following fix below:
1. Refactor DS code to use common method when setting UDDI spec version.
2. Update DS outbound class to set correct UDDI spec version

Assign @alameluchidambaram @christophermay07 for peer review?

### CODE REVIEW

File: Product/Production/Gateway/DocumentSubmission_20/src/main/java/gov/hhs/fha/nhinc/docsubmission/_20/entity/deferred/response/EntityDocSubmissionDeferredResponseImpl_g1.java [hunk 1]
```diff
@@ -59,10 +65,26 @@ public XDRAcknowledgementType provideAndRegisterDocumentSetBResponse(
...
+     * @return
+     */
+    private XDRAcknowledgementType provideAndRegisterDocumentSetResponse(RegistryResponseType request,
+            AssertionType assertion, NhinTargetCommunitiesType targets){
+        XDRAcknowledgementType response = null;
+        try{
+            DocSubmissionUtils.getInstance().setTargetCommunitiesVersion(targets, UDDI_SPEC_VERSION.SPEC_2_0);
+            response = outboundDocSubmissionResponse.provideAndRegisterDocumentSetBAsyncResponse(request,assertion,targets);
+        }catch(Exception e){
+            LOG.error("Failed to send request to Nwhin ",e);
```
Comment (christophermay07) [inline Product/Production/Gateway/DocumentSubmission_20/src/main/java/gov/hhs/fha/nhinc/docsubmission/_20/entity/deferred/response/EntityDocSubmissionDeferredResponseImpl_g1.java:86]: Can you log exceptions using format ("message: {}", e.getLocalizedMessage(), e)?

File: Product/Production/Gateway/DocumentSubmission_20/src/main/java/gov/hhs/fha/nhinc/docsubmission/_20/entity/deferred/request/EntityDocSubmissionDeferredRequestImpl_g1.java [hunk 2]
```diff
@@ -62,11 +69,31 @@ public XDRAcknowledgementType provideAndRegisterDocumentSetBRequest(
...
+    private XDRAcknowledgementType provideAndRegisterDocumentSetBAsyncRequest(
+            ProvideAndRegisterDocumentSetRequestType request, AssertionType assertion,
+            NhinTargetCommunitiesType targets, UrlInfoType urlInfo) {
+        XDRAcknowledgementType response = null;
+        try {
+            DocSubmissionUtils.getInstance().setTargetCommunitiesVersion(targets, UDDI_SPEC_VERSION.SPEC_2_0);
+            response = outboundDocSubmissionRequest.provideAndRegisterDocumentSetBAsyncRequest(request, assertion,
+                    targets, urlInfo);
+        } catch (Exception e) {
+            LOG.error("Failed to send request to Nwhin.", e);
```
Comment (christophermay07) [inline Product/Production/Gateway/DocumentSubmission_20/src/main/java/gov/hhs/fha/nhinc/docsubmission/_20/entity/deferred/request/EntityDocSubmissionDeferredRequestImpl_g1.java:95]: Same exception formatting comment as below.

**Comment by christophermay07** [comment]
I've reviewed this PR, endorsing for merge without further comment.

**Comment by alameluchidambaram** [comment]
I have reviewed this PR and good with merge.

**Pull request merged**